### PR TITLE
Added cadre details for the profile page

### PR DESCRIPTION
**Pull request by sujithsojan**

### CODE REVIEW

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 1]
```diff
@@ -616,20 +707,10 @@
             <mat-icon>info_outline</mat-icon>
           </div>
         </div>
-        <div class="flex flex-row justify-between">
-          <div class="flex flex-row gap-4 items-center" *ngIf="!editName"> 
-            <div class="name">{{ portalProfile?.personalDetails?.firstname | titlecase}}</div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: why this code has been removed, it should be there

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 2]
```diff
@@ -677,11 +758,6 @@
                 </ng-container>
               </ng-container>
             </div>
-            <div *ngIf="isMentor" class="flex flex-row gap-2 items-center view-mentor-profile-mob" (click)="viewMentorProfile()">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: why this code has been removed, it should be there

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.scss [hunk 3]
```diff
@@ -115,16 +126,6 @@ $border-color: rgba($color: #000000, $alpha: 0.16);
   cursor: pointer;
 }
 
-.blue-background-outline {
-  border-radius: 20px;
-  padding: 4px 16px;
-  border: 1px solid $text-primary;
-  color: white;
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.scss]: why this code has been removed, it should be there

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 4]
```diff
@@ -32,7 +32,8 @@ import { DesignationRequestComponent } from '../../components/designation-reques
 import { HomePageService } from 'src/app/services/home-page.service'
 import { RejectionReasonPopupComponent } from '../../components/rejection-reason-popup/rejection-reason-popup.component'
 import { ConfirmDialogComponent } from '@sunbird-cb/collection/src/lib/_common/confirm-dialog/confirm-dialog.component'
-import { environment } from 'src/environments/environment'
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts:35]: why this code has been removed, it should be there

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 5]
```diff
@@ -246,10 +277,6 @@ export class ProfileViewComponent implements OnInit, AfterViewInit, OnDestroy {
     this.pageData = this.route.parent && this.route.parent.snapshot.data.pageData.data
     this.currentUser = this.configService && this.configService.userProfile
 
-    if (this.configService && this.configService.userRoles) {
-      this.isMentor = this.configService.userRoles.has('MENTOR') ? true : false
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts]: why this code has been removed, it should be there

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 6]
```diff
@@ -1278,12 +1403,6 @@ export class ProfileViewComponent implements OnInit, AfterViewInit, OnDestroy {
     })
   }
 
-  viewMentorProfile() {
-    // window.location.href = 'https://portal.karmayogi.nic.in/mentorship'
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts]: why this code has been removed, it should be there

Comment (sujithsojan) [reply project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts]: Added all changes

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 7]
```diff
@@ -450,24 +450,115 @@
...
+              <div class="info-div" *ngIf="otherDetailsForm.get('typeOfCivilService').value && otherDetailsForm.get('isCadre').value || !editDetails || editingAsWhole">
+                <div class="label">{{ 'profileInfo.serviceType' | translate }}<span class="asterisk" style="color: red;"> *</span></div>
+                
+                <div class="info-value" *ngIf="!editDetails">
+                  <span class="flex flex-row gap-2 items-center">
+                    {{ portalProfile?.cadreDetails?.civilServiceName }}
+                  </span>
+                </div>
+              
+                <div class="flex flex-row gap-2" *ngIf="editDetails">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html:501]: Request for Service , we are not going to implement, so we can comment this

Comment (sujithsojan) [reply project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html:501]: Ok

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 8]
```diff
@@ -299,6 +332,79 @@ export class ProfileViewComponent implements OnInit, AfterViewInit, OnDestroy {
...
+    if (batchControl) { batchControl.reset() }
+    if (cadreControllingAuthorityControl) { cadreControllingAuthorityControl.reset() }
+
+    this.serviceType = this.civilServiceData.civilServiceTypeList.find((element: any) => element.name === event)
+    if (this.serviceType) {
+      this.serviceListData = this.serviceType.serviceList
+      this.serviceName = this.serviceListData.map((service: any) => service.name)
+      this.serviceId = this.serviceType.id
+    } else {
+      console.log('Service Type not found')
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts]: remove console

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 9]
```diff
@@ -299,6 +332,79 @@ export class ProfileViewComponent implements OnInit, AfterViewInit, OnDestroy {
...
+    const cadreControllingAuthorityControl = this.otherDetailsForm.get('cadreControllingAuthority')
+
+    if (cadreControllingAuthorityControl) { cadreControllingAuthorityControl.reset() }
+    this.profileService.fetchCadre().subscribe({
+      next: response => {
+        this.civilServiceData = response.result.response.value.civilServiceType
+    this.civilServiceTypes = this.civilServiceData.civilServiceTypeList.map((service: any) => service.name)
+      },
+      error: err => {
+        console.error('Error fetching cadre data:', err)
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts]: please remove console or disable lint

**Pull request merged**